funcframework/framework.go

```diff
@@ -65,7 +65,7 @@ func RegisterEventFunction(path string, fn interface{}) {
 func RegisterCloudEventFunction(path string, fn interface{}) {
 	fnCE, ok := fn.(func(cloudevents.Event))
 	if !ok {
-		log.Fatal("expected function to have signature func(cloudevents.Event), got %s", reflect.TypeOf(fn))
+		log.Fatalf("expected function to have signature func(cloudevents.Event), got %s", reflect.TypeOf(fn))
```

Comment:
Is there a particular reason to use `interface{}` rather than the actual type? https://play.golang.org/p/0M--vNNC4Nh

Reply:
Yes. When we generate the main function, it's dynamic (though perhaps it needn't be) and so the input needs to be generic. See https://github.com/GoogleCloudPlatform/buildpacks/blob/master/cmd/go/functions_framework/template.go#L40.

It might make more sense to have separate main files for each function type, which would provide a static interface and would catch signature errors at build time....I'll play around with it. Would be a separate PR, regardless.

Reply:
At build time, the user's function should have the right type, though, right?

Reply:
Yes, but if you follow that link, you'll notice that it has code paths to register the function as _all_ types. To do as you suggest without changes on that side yields compilation errors, as it tries to compile a function signature type that does not match. 

Like I said, there are changes that could be made there, but I would much prefer to do them separate to this.

Reply:
Since the main file will need changes anyway due to the error handling improvments, doing this here too.